developer/images/dependencies-update/update.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR="$(
+  cd "$(dirname "$0")" >/dev/null
+  pwd
+)"
+PROJECT_DIR="$(
+  cd "$SCRIPT_DIR/../../.." >/dev/null || exit 1
+  pwd
+)"
+export PROJECT_DIR
+
+usage() {
+  echo "
+Usage:
+    ${0##*/} [options]
+
+Upgrade Pipeline Service dependencies.
+
+Optional arguments:
+    -t, --task TASKNAME
+        Only run the selected task. Can be repeated to run multiple tasks.
+        TASKNAME must be in [$(echo "${DEFAULT_TASKS[@]}" | sed 's: :, :g')].
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    ${0##*/}
+" >&2
+}
+
+parse_args() {
+  mapfile -t DEFAULT_TASKS < <(find "$SCRIPT_DIR/tasks" -type f -name \*.sh -exec basename {} \; | sed 's:...$::')
+  TASKS=()
+  while [[ $# -gt 0 ]]; do
+    case $1 in
+    -t | --task)
+      shift
+      TASKS+=("$1")
+      ;;
+    -d | --debug)
+      set -x
+      DEBUG="--debug"
+      export DEBUG
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    --)
+      # End of arguments
+      break
+      ;;
+    *)
+      echo "Unknown argument: $1"
+      usage
+      exit 1
+      ;;
+    esac
+    shift
+  done
+}
+
+init() {
+  if [ -z "${TASKS[*]}" ]; then
+    TASKS=( "${DEFAULT_TASKS[@]}" )
+  fi
+  COMMIT_MSG="$TMPDIR/update_commit_msg.txt"
```

Comment:
When I manually executed `update.sh` without parametes, it reported error `TMPDIR: unbound variable`.  It needs to clarify exporting the value of `TMPDIR` before executing `update.sh` in the `usage` function.

Reply:
Fixed.